Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::White);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_debugtext_y / dpiScalingFactor, editorsv_connectionMessageFontSize, m_centerViewportDebugText.c_str(), true);
+        }
+
+        // Build the connection status string (just show client connected or disconnected status for now)
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();        
+        if (agentType != MultiplayerAgentType::Client)
+        {
+            return;
+        }
+
+        // Display the connection status to the top-left viewport
+        if (AzNetworking::INetworkInterface* networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName)))
+        {
+            AzNetworking::IConnectionSet& connectionSet = networkInterface->GetConnectionSet();
+            if (connectionSet.GetConnectionCount() > 0)
+            {
+                // Display the connection status (calling VisitConnections(), but there's only 1 since we're a client)
+                auto displayConnectionStatus = [](AzNetworking::IConnection& connection)
+                {
+                    AZStd::string connectionStatusText = AZStd::string::format("Operating as multiplayer Client: %s", GetEnumString(connection.GetConnectionState()));
+                    AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::White);
+                    AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                        editorsv_connectionStatus_topLeftBorderPadding_x, editorsv_connectionStatus_topLeftBorderPadding_y, editorsv_connectionMessageFontSize, 
+                        connectionStatusText.c_str(), false);
+                };
+                connectionSet.VisitConnections(displayConnectionStatus);
+            }
+            else
+            {
+                // If we're a client yet are lacking a connection then we've been unintentionally disconnected
+                // Display a disconnect message in the viewport
+                AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::Red);
+                AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                    editorsv_connectionStatus_topLeftBorderPadding_x, editorsv_connectionStatus_topLeftBorderPadding_y, editorsv_connectionMessageFontSize, 
+                    "Multiplayer Client Disconnected!", false);
```

Comment:
Nit: May worth defining a constant for the error message.

Reply:
Personally when it comes to error messages I prefer to keep them local to the message for searchability.